### PR TITLE
README: update development section

### LINKED ISSUE #269: Clear up installation instructions
We should update the installation instructions to

1. work, and
2. maybe even use glide for reproducible builds.

### PR DESCRIPTION
**Pull request by jzelinskie**
Fixes #269.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,16 +57,42 @@ By verifying the infohash, Quay can be sure that only their content is being sha
...
+
+[glide]: https://glide.sh
+
+#### Testing
+
+The following will run all tests and benchmarks.
+Removing `-bench` will just run unit tests.
+
+```sh
+$ go test -bench $(glide novendor | grep -v contrib)
```
Comment (mrd0ll4r) [inline README.md]: `-bench` needs a filter, doesn't it? Something like `-bench .` ?

File: README.md [hunk 2]
```diff
@@ -57,16 +57,42 @@ By verifying the infohash, Quay can be sure that only their content is being sha
 
 ### Getting Started
 
-In order to compile the project, the [latest stable version of Go] and a [working Go environment] are required.
+#### Building from HEAD
+
+In order to compile the project, the [latest stable version of Go] and a knowledge of a [working Go environment] are required.
```
Comment (mrd0ll4r) [inline README.md]: ...and ~a~ knowledge...

File: README.md (hunk 1 above)
Comment (jzelinskie) [inline README.md]: `$(glide novendor)` lists all the go packages in the directory.

This worked when I tried it ¯\_(ツ)_/¯

Comment (mrd0ll4r) [inline README.md]: It really does, I'm baffled...

**Pull request merged**